tests/lax_numpy_test.py

```diff
@@ -2558,6 +2558,26 @@ def testPrecision(self):
         partial(lnp.inner, precision=HIGHEST),
         ones_1d, ones_1d)
 
+  @parameterized.named_parameters(
+      jtu.cases_from_list(
+        {"testcase_name": ("_shape={}_dtype={}").format(shape, dtype),
+         "shape": shape,
+         "dtype": dtype, "rng_factory": rng_factory}
+        for shape in [(10,), (10, 15), (10, 15, 20)]
+        for dtype in inexact_dtypes
+        for rng_factory in [jtu.rand_default]))
+  def testGradient(self, shape, dtype, rng_factory):
+    rng = rng_factory()
+    args_maker = self._GetArgsMaker(rng, [shape], [dtype])
+    ndim = len(shape)
+    for num_axes in range(ndim):
+      for axis in itertools.combinations(range(ndim), num_axes):
```

Comment:
It would be slightly preferable to move these loops into the `named_parameters` comprehension above. That should make test cases point more directly to failing tests if/when something goes wrong.

(Control flow inside test cases can be a little challenging to debug, because you usually only have access to the failing test message which doesn't tell you at which iteration the test failed.)